src/lib/protocols/credentials/messages/CredentialAckMessage.ts

```diff
@@ -0,0 +1,21 @@
+import { Equals } from 'class-validator';
+import { AgentMessage } from '../../../agent/AgentMessage';
+import { MessageType } from './MessageType';
+
+interface CredentialAckMessageOptions {
+  id?: string;
+}
+
+export class CredentialAckMessage extends AgentMessage {
+  public constructor(options: CredentialAckMessageOptions) {
```

Comment:
That's a good point and I agree. However, if I do that we would need to change the logic to something like this:
```javascript
    if (options) {
      this.id = options.id ?? this.generateId();
    } else {
      this.id = this.generateId();
    }
```

And I don't like that. I suggest to keep it for now and maybe change later if we see other cases like this one.

Reply:
Good point. We definitely don't want that.

Ideally we should get rid of the `if (options)` in all classes. There are others that have the same problem. `class-transformer` calls `new XXX` while it should use reflection directly. See [the issue (and my comment)](https://github.com/typestack/class-transformer/issues/132). I'll try to take a look as it will be of huge benefit to us.